mesh/src/main/java/no/nordicsemi/android/mesh/transport/GenericLocationGlobalSet.java

```diff
@@ -0,0 +1,97 @@
+package no.nordicsemi.android.mesh.transport;
+
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import no.nordicsemi.android.mesh.ApplicationKey;
+import no.nordicsemi.android.mesh.data.GlobalAltitude;
+import no.nordicsemi.android.mesh.data.GlobalLatitude;
+import no.nordicsemi.android.mesh.data.GlobalLongitude;
+import no.nordicsemi.android.mesh.opcodes.ApplicationMessageOpCodes;
+import no.nordicsemi.android.mesh.utils.SecureUtils;
+
+/**
+ * To be used as a wrapper class for when creating the GenericLocationGlobalSet message.
+ */
+public final class GenericLocationGlobalSet extends ApplicationMessage {
+    private static final String TAG = GenericLocationGlobalSet.class.getSimpleName();
+    private static final int OP_CODE = ApplicationMessageOpCodes.GENERIC_LOCATION_GLOBAL_SET;
+    private static final int GENERIC_LOCATION_GLOBAL_SET_LENGTH = 10;
+    private GlobalLatitude latitude;
+    private GlobalLongitude longitude;
+    private GlobalAltitude altitude;
+
+    /**
+     * Constructs GenericLocationGlobalSet message.
+     *
+     * @param appKey    {@link ApplicationKey} key for this message
+     * @param latitude  {@link GlobalLatitude} global latitude
+     * @param longitude {@link GlobalLongitude} global longitude
+     * @param altitude  {@link GlobalAltitude} global altitude
+     */
+    public GenericLocationGlobalSet(
+            @NonNull final ApplicationKey appKey,
+            @NonNull final GlobalLatitude latitude,
+            @NonNull final GlobalLongitude longitude,
+            @NonNull final GlobalAltitude altitude) {
+        super(appKey);
+        this.latitude = latitude;
+        this.longitude = longitude;
+        this.altitude = altitude;
+        assembleMessageParameters();
+    }
+
+    @Override
+    void assembleMessageParameters() {
+        this.mAid = SecureUtils.calculateK4(mAppKey.getKey());
+        final ByteBuffer buffer = ByteBuffer.allocate(GENERIC_LOCATION_GLOBAL_SET_LENGTH).order(ByteOrder.LITTLE_ENDIAN);
+        Log.v(TAG, "Creating message");
+        Log.v(TAG, "Latitude: " + this.latitude);
```

Comment:
Wouldn't this print Latitude: GlobalLatitude.....? A bit redundant.

Reply:
yes, redundant: 
GenericLocationGlobalSet: Latitude: GlobalLatitude.Coordinate encodedValue: 7fffffff position: 90.0